proxy.go

```diff
@@ -60,13 +61,19 @@ func newReverseProxy(cfgCp *config.ConnectionPool) *reverseProxy {
 			return dialer.DialContext(ctx, network, addr)
 		},
 		ForceAttemptHTTP2:     true,
-		MaxIdleConns:          cfgCp.MaxIdleConns,
-		MaxIdleConnsPerHost:   cfgCp.MaxIdleConnsPerHost,
+		MaxIdleConns:          cfgCp.ConnectionPool.MaxIdleConns,
+		MaxIdleConnsPerHost:   cfgCp.ConnectionPool.MaxIdleConnsPerHost,
 		IdleConnTimeout:       90 * time.Second,
 		TLSHandshakeTimeout:   10 * time.Second,
 		ExpectContinueTimeout: 1 * time.Second,
 	}
 
+	if cfgCp.InsecureSkipVerify {
```

Comment:
Maybe we can just?

```suggestion
transport.TLSClientConfig = &tls.Config{
	InsecureSkipVerify: cfgCp.InsecureSkipVerify, //nolint:gosec
}
```

Not sure if creating the TLSClientConfig object changes the behavior though, in that case, just ignore this!

Reply:
I had this code as you suggested before.
I've tried to dig on this one into the source. There's a one warning about HTTP2. I have no confidence in committing it.